ggml/src/ggml-cuda/argmax.cu

```diff
@@ -1,57 +1,69 @@
-#include "common.cuh"
+#include <algorithm>
+#include <cstdint>
+
 #include "argmax.cuh"
+#include "common.cuh"
 #include "sum.cuh"
 
-#include <cstdint>
+static __global__ void argmax_f32(const float * __restrict__ x, int32_t * __restrict__ dst, const int64_t ncols) {
+    const int64_t row = blockIdx.x;
 
-static __global__ void argmax_f32(
-    const float * x, int32_t * dst, const int64_t ncols, const int64_t nrows) {
+    float maxval = -FLT_MAX;
+    int   argmax = -1;
```

Comment:
Looking at the code again, I think either 64 bit should be used for the `ne00` dimension or there should be an assert that 32 bit is enough.

Reply:
The output is int32, so it would definitely not worth with `ne00` larger than `INT_MAX`. In that case it might make more sense to add the assert to `ggml_argmax` instead. Other arg* functions will have the same issue.